include/bout/mesh.hxx

```diff
@@ -216,6 +216,9 @@ class Mesh {
   /// @returns zero always. 
   int get(Vector3D &var, const std::string &name, BoutReal def=0.0);
 
+  /// Test if input source was a grid file
+  bool isSourceFile() const;
```

Comment:
To be honest, I struggle to read this as anything other than "is `mesh` a 'source file'?", rather than "is the `mesh` source a file?".

On the other hand, I'm also struggling to think of a better name! `isSourceFromFile` maybe?

Reply:
IsTheSourceAFile?

Reply:
I quite like the latter. `mesh->isDataSourceGridFile()` reads pretty clearly to me.